jupyter_core/application.py

```diff
@@ -143,6 +146,20 @@ def ask():
         with open(config_file, mode='w') as f:
             f.write(config_text)
     
+    def migrate_config(self):
+        """Migrate config/data from IPython 3"""
+        if os.path.exists(os.path.join(self.config_dir, 'migrated')):
+            # already migrated
+            return
+
+        ipdir = os.environ.get('IPYTHONDIR') or os.path.expanduser('~/.ipython')
```

Comment:
We're importing/defining `get_ipython_dir` in the migrate module, should this import and use that rather than copying only the fallback logic?